pkg/chunk/fifo_cache.go

```diff
@@ -1,20 +1,79 @@
 package chunk
```

Comment:
Mostly a note/question, it doesn't look like FifoCache is related to the chunk. Could it be pulled out to a different package in the future?